samples/bluetooth/iso_time_sync/src/bis_receiver.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** BIS receiver implementation for the time sync sample
+ *
+ * This file implements a device that syncs to a given BIS index.
+ * When disconnected, it tries to sync to it again.
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/iso.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <sdc_hci_vs.h>
+#include <sdc_hci.h>
+#include "iso_time_sync.h"
+
+#define ADV_NAME_STR_MAX_LEN (sizeof(CONFIG_BT_DEVICE_NAME))
+
+static K_SEM_DEFINE(sem_per_adv_info_valid, 0, 1);
+static K_SEM_DEFINE(sem_big_info_valid, 0, 1);
+static K_SEM_DEFINE(sem_per_adv_sync_lost, 0, 1);
+
+static bool syncing_or_synced_to_periodic_adv;
+static bool syncing_or_synced_to_big;
+
+static bt_addr_le_t adv_addr;
+static uint8_t adv_set_id;
+
+static bool adv_data_parse_cb(struct bt_data *data, void *user_data)
+{
+	char *name = user_data;
+	uint8_t len;
+
+	switch (data->type) {
+	case BT_DATA_NAME_SHORTENED:
+	case BT_DATA_NAME_COMPLETE:
+		len = MIN(data->data_len, ADV_NAME_STR_MAX_LEN - 1);
+		memcpy(name, data->data, len);
+		name[len] = '\0';
+		return false;
+	default:
+		return true;
+	}
+}
+
+static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *buf)
```

Comment:
Could you use NCS scanning module and set name filter?

Reply:
I looked through the code to see how much it will simplify things. I don't think it is worth doing:

Before there is less initialization code and slightly more code in the callback
```c
static bool adv_data_parse_cb(struct bt_data *data, void *user_data)
{
	char *name = user_data;
	uint8_t len;

	switch (data->type) {
	case BT_DATA_NAME_SHORTENED:
	case BT_DATA_NAME_COMPLETE:
		len = MIN(data->data_len, ADV_NAME_STR_MAX_LEN - 1);
		memcpy(name, data->data, len);
		name[len] = '\0';
		return false;
	default:
		return true;
	}
}


static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_simple *buf) 
{
	char name_str[ADV_NAME_STR_MAX_LEN] = {0};

	bt_data_parse(buf, adv_data_parse_cb, name_str);

	if (strncmp(name_str, CONFIG_BT_DEVICE_NAME, ADV_NAME_STR_MAX_LEN) != 0) {
		return;
	}
	/* ... */
}

	err = bt_le_scan_start(
			BT_LE_SCAN_PARAM(BT_LE_SCAN_TYPE_PASSIVE, BT_LE_SCAN_OPT_FILTER_DUPLICATE,
					 BT_GAP_SCAN_FAST_INTERVAL, BT_GAP_SCAN_FAST_INTERVAL),
			NULL);
```

After there is more initialization code and only slightly less code in the callback
```c
static void scan_filter_match(struct bt_scan_device_info *device_info,
			      struct bt_scan_filter_match *filter_match,
			      bool connectable) {
		/* ... */
}

		static const struct bt_le_scan_param sp = {
				.type = BT_LE_SCAN_TYPE_ACTIVE,
				.options = BT_LE_SCAN_OPT_FILTER_DUPLICATE,
				.interval = BT_GAP_SCAN_FAST_INTERVAL,
				.window = BT_GAP_SCAN_FAST_WINDOW,
		};

		struct bt_scan_init_param scan_init = {
				.connect_if_match = false,
				.scan_param = &sp,
		};

		bt_scan_init(&scan_init);

		err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_NAME, CONFIG_BT_DEVICE_NAME);
		if (err) {
			LOG_ERR("Name filter cannot be added (err %d)", err);
			break;
		}

		err = bt_scan_start(BT_SCAN_TYPE_SCAN_ACTIVE);
		if (err) {
				LOG_ERR("Cannot start scanning (err %d)", err);
				return err;
		}
```

That is, in total the amount of code is almost the same.

Reply:
If you use default scan amd connection params. Then it will be just:
```
		struct bt_scan_init_param scan_init = {
				.connect_if_match = true,
		};

		bt_scan_init(&scan_init);

		err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_NAME, name);
		if (err) {
			LOG_ERR("Name filter cannot be added (err %d)", err);
			break;
		}

		err = bt_scan_start(BT_SCAN_TYPE_SCAN_ACTIVE);
		if (err) {
				LOG_ERR("Cannot start scanning (err %d)", err);
				return err;
		}
```

and connection will be established automatically inside module

Reply:
It is intentional that we are not using the default scan parameters, as we need to use continuous scanning.

Also, by keeping it as is, we can keep the code for the bis_receiver and cis_peripheral identical.

Therefore I would like to keep it as is.